S3 static walkthroughs (#8)
LaunchCodeEducation/aws-basics: content/s3/walkthrough/static-websites/angular-static-website/_index.md

- `Index Document`: The Index Document you will be using is `index.html` that you uploaded earlier in this walkthrough.
- `Error Document`: This option you will leave as defaulted.
- `Redirection Rules`: This option you will leave as defaulted.

![Edit Static Website Hosting Expanded View](pictures/edit-static-website-expanded.png?classes=border)

After you have selected and filled in the correct values scroll to the bottom of the page and click the `Save Changes` button.

### Validation

After saving the above changes you should be able to scroll to the bottom of your `Properties` tab and see the changes reflected under `Static Website Hosting`.

![Static Website Hosting Options Saved](pictures/static-website-hosting-validation.png?classes=border)

You can also now see the endpoint for your static website. In the above screenshot the bucket endpoint is:
- `http://my-first-bucket-john.s3-website-us-east-1.amazonaws.com`

If you click on the link your browser should open up a new window. You will most likely receive the following error on that new page:

![403 forbidden whitepage](pictures/403-forbidden-whitepage.png?classes=border)

> This is there is no bucket policy attached to the `orbit-report-bucket`. You will need to attach a new bucket policy to this bucket in order to make the objects available to the public.

Reviewer: ```suggestion
No bucket policy was attached to the `orbit-report-bucket`. You will need to attach a new bucket policy to this bucket in order to make the objects available to the public.
```

Click on the `Edit` button.

This will open up a new view so that you are able to attach a policy to the bucket.

You will be using the bucket policy provided in this article: [AWS S3 Security Permissions Document](https://docs.aws.amazon.com/AmazonS3/latest/userguide/WebsiteAccessPermissionsReqd.html)

You can find the raw json below:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "PublicReadGetObject",
            "Effect": "Allow",
            "Principal": "*",
            "Action": [
                "s3:GetObject"
            ],
            "Resource": [
                "arn:aws:s3:::Bucket-Name/*"
            ]
        }
    ]
}
```

Add the above `Policy` to the empty field as shown in the screenshot below, replacing `Bucket-Name` with the name of your bucket:

![Public Bucket Polcy S3 Bucket](pictures/attach-bucket-policy.png?classes=border)

{{% notice warning %}}
You will need to add the name of your bucket under the "Resource" section in order for the policy to work. Without a valid bucket name you will most likely receive an error if you try to save changes.
{{% /notice %}}

Click on the `Save changes` button.

### Validation

> Now that you have attached a `Policy` to your bucket you should now be able to access your static website using the link found under the `Static Website` section under your `Properties` tab!

Reviewer: ```suggestion
Now that you have attached a `Policy` to your bucket, you should now be able to access your static website using the link found under the `Static Website` section under your `Properties` tab!
```
